CHANGELOG.md

```diff
@@ -9,6 +9,10 @@ Changelog
   addresses
   [#319](https://github.com/bugsnag/bugsnag-cocoa/pull/319)
 
+* Add `fatalError` and other assertion failure messages in reports for
+  Swift 4.2 apps
```

Comment:
It'd be worth noting that we can't capture messages <16 characters

Reply:
Updated, also updated the commit with some addition information about the limitations:

    In the new Xcode 10 build system, Swift object register values have the
    top bit used as a flag. This change strips the flag while not losing
    anything relevant to us in our quest to see error messages for assertion
    failures.

    This technique does not capture messages which are less than 16
    characters, as short strings are stored as raw char arrays on the stack
    rather than being allocated. (See WWDC 2018 #401 for more info on new
    string optimizations)

    While it is possible to check for char arrays as well as pointers when
    searching for notable address values, sweeping up local variables has a
    likely chance of capturing unintended data as well from the surrounding
    code, some of which may be sensitive. It is also not guaranteed that the
    value would still be on the stack after the message is logged, so it is
    possible to get only unrelated string values as the message.

    In the current Swift stdlib, the following messages passed to
    fatalError, preconditionFailure, and precondition (and their internal
    func counterparts) are less than 16 characters:

    * empty string
    * `unavailable`
    * `not implemented`
    * `abstract method`
    * `unknown value`
    * `invalid count` (where a dictionary contains < 0 items(?))
    * `invalid index` (where a dictionary ceases to be a dictionary)
    * `don't touch me` (from SpriteKit)
    * `close() failed` (from the private Subprocess implementation)

    The vast majority have more meaningful messages.

And a relevant quote from the linked wwdc session:

> If the string fits within 15 bytes then the actual string is represented directly in the string type without going to the heap to allocate a separate buffer to represent the string.